src/oc_erchef/apps/oc_chef_wm/src/oc_chef_data_collector.erl

```diff
@@ -0,0 +1,60 @@
+-module(oc_chef_data_collector).
+
+-include("oc_chef_wm.hrl").
+
+-export([notify/2]).
+
+-spec notify(Req :: wm_req(), State :: #base_state{}) -> ok.
+notify(Req, #base_state{reqid = ReqId, resource_state = ResourceState, resource_mod = ResourceMod} = State) when
+    is_record(ResourceState, acl_state);
+    is_record(ResourceState, association_state);
+    is_record(ResourceState, client_state);
+    %% Skip controls because oc_chef_action can't extract them yet
+    %% is_record(ResourceState, control_state);
+    is_record(ResourceState, cookbook_state);
+    is_record(ResourceState, data_state);
+    is_record(ResourceState, environment_state);
+    is_record(ResourceState, group_state);
+    is_record(ResourceState, node_state);
+    is_record(ResourceState, organization_state);
+    is_record(ResourceState, role_state);
+    is_record(ResourceState, user_state);
+    is_record(ResourceState, key_state) ->
+        case {wrq:method(Req), req_success(Req), ResourceMod} of
+            %% Skip GET because we're not updating anything
+            {'GET', _, _} ->
+                ok;
+            %% authenticate_user should not be an action.
+            {_, _, oc_chef_wm_authenticate_user} ->
+                ok;
+            %% Notify any other 2XX message for matching resources
+            {_, true, _} ->
+                Msg = oc_chef_action:create_message(Req, State, true),
+                stats_hero:ctime(ReqId, {data_collector, update}, fun() -> data_collector_http:post("/", Msg) end);
```

Comment:
Uff. Sorry about this, but I forgot to mention that I think in order to add a new upstream like this (rather than using an existing upstream), you'll need to modify the functions here:

https://github.com/chef/chef-server/blob/master/src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_base.erl#L642-L666